internal/controller/capicluster_controller.go

```diff
@@ -97,3 +104,14 @@ func (r *CapiClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		).
 		Complete(r)
 }
+
+// ControllerRateLimiter returns a rate limiter to control the rate of reconciliation.
+// Taken from https://github.com/kubernetes/client-go/blob/20732a1bc198ab57de644af498fa75e73fa44c08/util/workqueue/default_rate_limiters.go#L39-L45
```

Comment:
Can you explain why you add this rate limiter?

Reply:
We though the rate limiter would help reduce the amount of incoming request and therefore reduce the amount of errors, but it doesn't as it only act on internal retries. So we abandoned this idea.